rio-labs/rio
# Replace string literals with ENUM types

_DO NOT MERGE YET - NOT FINAL_

### What does it do?

This pull request replaces the existing Literal string type annotations with an Enum type. This change is aimed at improving the maintainability, readability, and robustness of the codebase.

### Why is it needed?

Previously, the codebase used Literal string type annotations to define a set of valid string values for specific variables and function parameters. This approach, while functional, has several limitations that can be addressed by using Enum types:

**Improved Readability:**

Enums provide a clear, self-documenting way to represent a set of constant values. By using descriptive names for enum members, the code becomes more understandable and easier to read.

**Type Safety:**

Enums enforce type safety more rigorously than Literal strings. They ensure that only predefined values can be assigned, reducing the risk of runtime errors caused by invalid string literals.

**Enhanced Maintainability:**

Centralizing the definition of valid values in an Enum makes it easier to manage and update the set of possible values. Changes to the valid values need to be made in only one place, minimizing the risk of inconsistencies.

**Auto-Completion and IDE Support:**

Modern IDEs provide better support for enums, offering auto-completion and inline documentation, which can speed up development and reduce errors.

### How to test it?

Just install it and run any predefined template. Nothing should've changed. But I have not yet understood how the translation of width/height values into JS/HTML/CSS work !

### Example

Previously:

```py
    # [...]
    width: float | Literal["natural", "grow"] = "natural"
    height: float | Literal["natural", "grow"] = "natural"
```

Now:
```py
    # [...]
    width: float | ExpandStrategy = ExpandStrategy.NATURAL
    height: float | ExpandStrategy = ExpandStrategy.NATURAL
```
